### DISCUSSION TITLE
монастырь // маг огня // послушник // KDF

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#502] [General]
https://github.com/dosinabox/g2nr_unofficial_update/issues?q=label%3AKDF+is%3Aclosed

**Comment by Kor-Angar** [comment]
есть предложение удалять из сундука Игараца документы при вступление в Монастырь. вместо них ему в сундук можно положить например дурь

**Reply by Kor-Angar** [reply]
таким образом Послушник может украсть его ключ и открыть его сундук, получив какую то плюшки.

**Reply by Kor-Angar** [reply]
а он не уникальный :-( в него через скрипты ничего не вставишь без юниона

```D
		childs6666=int:0
		[% oCMobContainer:oCMobInter:oCMOB:zCVob 64513 7618]
			pack=int:0
			presetName=string:
			bbox3DWS=rawFloat:47953.6445 4992.00098 21682.1895 48058.2148 5069.46875 21787.4258 
			trafoOSToWSRot=raw:f08a30bf00000000436339bf000000000000803f000000004463393f00000000f38a30bf
			trafoOSToWSPos=vec3:48007.6992 4992 21736.9063
			vobName=string:
			visual=string:CHESTBIG_NW_NORMAL_LOCKED.MDS
			showVisual=bool:1
			visualCamAlign=enum:0
			visualAniMode=enum:0
			visualAniModeStrength=float:0
			vobFarClipZScale=float:1
			cdStatic=bool:1
			cdDyn=bool:1
			staticVob=bool:1
			dynShadow=enum:0
			zbias=int:0
			isAmbient=bool:0
			[visual zCModel 0 7619]
			[]
			[ai % 0 0]
			[]
			focusName=string:MOBNAME_IGARAZ
			hitpoints=int:10
			damage=int:0
			moveable=bool:0
			takeable=bool:0
			focusOverride=bool:0
			soundMaterial=enum:0
			visualDestroyed=string:
			owner=string:NOV_601_Igaraz
			ownerGuild=string:
			isDestroyed=bool:0
			stateNum=int:1
			triggerTarget=string:
			useWithItem=string:
			conditionFunc=string:
			onStateFunc=string:
			rewind=bool:0
			locked=bool:-1
			keyInstance=string:ItKe_IgarazChest_mis
			pickLockStr=string:
			contains=string:ItWr_BabosDocs_MIS
		[]
```

**Reply by Kor-Angar** [reply]
с другой стороны можно внутри функции "связки бумаг" сделать проверку и давать какие то другие документы. например какой то рецепт или несколько свитков

```D
func void Use_BabosDocs()
{
    if (hero.guild == GIL_NOV)
    || (hero.guild == GIL_KDF)
    {
        CreateInvItems(self,ITWr_Addon_Joint_01,1);
    }
    else
    {
        Use_BabosDocs_old();
    };
};
```

**Reply by dosinabox** [reply]
одну картинку, без письма)

**Reply by Kor-Angar** [reply]
там же связка бумаг. можно кстати картинку + карту со статуями инноса ;-)

**Reply by Kor-Angar** [reply]
> одну картинку, без письма)

@dosinabox   поправь. сейчас полноценная связка которая запускает квест не для мага в первой главе и даёт за него опыт и награду. 
![20230819183743_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/d3cde146-9aaf-4a39-94d1-ebb57e363f36)
![20230819183858_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/da08f7f0-f2bb-44fd-b193-7568dcefca07)

**Reply by dosinabox** [reply]
А как в оригинале было?

**Reply by Kor-Angar** [reply]
в оригинале данный  ключ  послушник у него украсть не мог.

**Reply by dosinabox** [reply]
Сундук Игараца теперь именной, а значит связку бумаг можно добавлять в третьей главе для нужных гильдий:
```cpp
if(C_WorldIsFixed(NEWWORLD_ZEN))
{
	Mob_CreateItems("CHEST_IGARAZ",ItWr_BabosDocs_MIS,1);
};
```

**Reply by dosinabox** [reply]
> в оригинале данный ключ послушник у него украсть не мог.

Но мог получить его во время испытания, побив или убив Игараца? Вижу, что ключ создается у него в инстанции.

**Reply by dosinabox** [reply]
Возвращаясь к вопросу.
- В оригинале и в моде ключ от сундука с бумагами создается в инвентаре Игараца с началом новой игры. Открыть сундук можно побив Игараца. Во время испытания он даже сам нападает на ГГ.
- В моде этот ключ также можно украсть с началом новой игры. Это не намеренное решение.

Что из этого является неправильным? Если послушник не должен получать доступ к этому сундуку и квесту в первой главе, то добавлять ключ в инвентарь Игараца следует только в третьей главе, проверяя гильдию ГГ. И тогда же делать его доступным для кражи.

**Reply by Kor-Angar** [reply]
в твоей версии сундук уникальный. соотв. на старте ты на скриптах можешь положить туда то какой то лут. а при старте 3 главы для наёмника и мента положить в этот сундук связку бумаг.

**Reply by dosinabox** [reply]
Могу. Но сначала эту связку бумаг надо из сундука убрать.

**Reply by Kor-Angar** [reply]
ты же скриптами можешь удалить при старте, а поместить туда какой то лут, например пару косяков травы.

**Reply by dosinabox** [reply]
Я хочу удалить документы в самом ZEN, а это обычно затягивается надолго. В качестве временного решения предлагаю добавлять ключ Игарацу только в третьей главе для нужных гильдий.

**Reply by dosinabox** [reply]
Функция Mob_CreateItems проверялась? Работает нормально?

**Reply by Kor-Angar** [reply]
а почему в качестве временного решения не удалять скриптами бумаги из сундука? :-)

> Функция Mob_CreateItems проверялась? Работает нормально?

в готики 1 работала. в НВ сейчас проверю

**Reply by Kor-Angar** [reply]
или скриптами без юнион предмет не удаляется из сундука?

**Reply by dosinabox** [reply]
> а почему в качестве временного решения не удалять скриптами бумаги из сундука? :-)

> или скриптами без юнион предмет не удаляется из сундука?

Это внешняя нестандартная функция. Я так не хочу.

**Reply by Kor-Angar** [reply]
тогда можно вернутся к тому что на уровне скрипта использования связки будут появляться разные документы.

**Reply by Kor-Angar** [reply]
```D
func void Use_BabosDocs()
{
    if (hero.guild == GIL_NOV)
    || (hero.guild == GIL_KDF)
    {
        CreateInvItems(self,ITWr_****,1);
    }
    else
    {
        ....
    };
};

```

**Reply by dosinabox** [reply]
Этот квест вообще ужасно кривой и недоделанный:
- ключ доступен для кражи с третьей главы, но с тела - с первой;
- кража недоступна магам, но ключ с тела может получить послушник;
- Бабо можно отдать документы, а потом спросить DIA_Babo_Kap3_Unhappy и начать уже выполненный квест;
- квест начинается сразу в диалоге DIA_Babo_Kap3_Unhappy, даже если игрок от него отказывается;
- недоступные диалоги DIA_Babo_Kap3_HaveYourDocs_KeepThem_Partner, DIA_Babo_Kap3_HaveYourDocs_KeepThem_Partner_KeepCalm, DIA_Babo_Kap3_HaveYourDocs_KeepThem_Partner_NothingToDo;
- можно украсть или отобрать ключ у Игараца, а потом купить второй за 300 золотых;
- ГГ не передает Бабо документы когда получает повышенную награду в 121 золотых;
- нет возможности "обговорить финансовую часть с Игарацем" (переменная Igaraz_IsPartner всегда FALSE);

**Comment by Kor-Angar** [comment]
при создание руны можно выводить её название

**Reply by Kor-Angar** [reply]
![20230212020310_1](https://user-images.githubusercontent.com/49514764/218286824-7ecfd991-bd05-4191-9662-a14fe0b872eb.jpg)

**Comment by Kor-Angar** [comment]
квест про кельи послушников. есть такое предложение, после того как 4ый послушник пойдёт подметать келию, замерять день и сдавать квест не тут же , а на следующий день.

сейчас герой может сдать квест до того как послушник даже приступит к подметанию келии.
![20230211221837_1](https://user-images.githubusercontent.com/49514764/218286863-0a2de936-ea23-4b72-b3a6-2106a26538d7.jpg)
![20230211221850_1](https://user-images.githubusercontent.com/49514764/218286874-cfd980e1-4bc7-4570-a3e0-570790491dc6.jpg)

**Comment by Kor-Angar** [comment]
вчера что то я бегал вокруг послушников по заданию на вступление в маги огня. надо еще раз к ним присмотреться:

https://www.youtube.com/watch?v=8ubyUERLyjM

1) менять расписание у Ульфа если он достиг "тайного места"
2) придумать сценарий если Ульф пришёл в пещеру с ГГ
3) Агону увеличить дальность преследование
4) ульфа добавить проверки на дистанцию от тайного места на инстанцию диалога с вариантом  "покурить"

**Comment by Kor-Angar** [comment]
топик про монастырь не создаётся в диалоге с Ксардасом, когда герой возвращается к нему не попав к Хагену и Ксардас рассказывает что можно в крайнем случае пойти в монастырь.

**Reply by dosinabox** [reply]
А какой должен создаваться?

**Comment by Kor-Angar** [comment]
феодал он подчёркивает свой социальный статус. в том числе через шмотки. потому что у него есть бабки, а у других нет. 
Корнелиус это специально обученая печатная машинка. ему надиктовывает текст, а он его записывает. перекладывает бумажки и так далее. ценный специалист. находится на довольствие. Агон это племянник губернатора из феодальной семьи. у его семьи есть земля и деньги. поколения за поколением.. а ты взяла из Агона сделала какого то нищеброда :-( где бабки? где дорогой меч? где дорогие шмотки? где та самая "благосклонность Инноса"? :-)

![20230213185338_1](https://user-images.githubusercontent.com/49514764/218775423-847da66d-ca2c-45d4-a670-821681290798.jpg)
![20230213185306_1](https://user-images.githubusercontent.com/49514764/218775469-2c1d6485-df55-4d11-bc14-807c8072a76b.jpg)

**Reply by Kor-Angar** [reply]
заметьте это не капитализм, где торгаш может прийти и заплатить 10 000 $ за ужин в ресторане или купить бизнес-джет. Там, в феодализме ценность была земля которая была вся распределена и её нельзя было просто "купить" взяв кредит. именно поэтому там феодал это феодал.

**Comment by Kor-Angar** [comment]
надо свитки добавить в монастырь из аддона. а то герой учить схему , а свиток можно купить или в городе или у магов воды.

**Reply by dosinabox** [reply]
при изучении новый свиток всегда создается в первую очередь у Горакса

**Reply by dosinabox** [reply]
B_CreateScrollForTrading

**Reply by Kor-Angar** [reply]
схема учится в мире аддона.

**Reply by Kor-Angar** [reply]
```D
		if(!Npc_IsDead(Cronos_ADW))
		{
			trader = Hlp_GetNpc(Cronos_ADW);
		};
```
кронос бессмертный.

**Reply by Kor-Angar** [reply]
вот тут  можно добавить свитков из аддона в продажу Гораксу, если герой маг огня.

```D
	if(ENTERED_ADDONWORLD == TRUE)
	{
		if(NW_KDW_Removed_Forever == FALSE)
		{
			B_RemoveNpc(KDW_1400_Addon_Saturas_NW);
			B_RemoveNpc(KDW_1401_Addon_Cronos_NW);
			B_RemoveNpc(KDW_1402_Addon_Nefarius_NW);
			B_RemoveNpc(KDW_1403_Addon_Myxir_NW);
			B_RemoveNpc(KDW_1404_Addon_Riordian_NW);
			B_RemoveNpc(KDW_1405_Addon_Merdarion_NW);
			B_RemoveNpc(PIR_1301_Addon_Skip_NW);
			if(FullNPCRemoval == TRUE)
			{
				NW_KDW_Removed_Forever = TRUE;
			};
		};
		if(SC_GotPortalTempelWalkthroughKey == FALSE)
		{
			Wld_InsertNpc(Stoneguardian_MerdarionsSchluessel,"NW_TROLLAREA_PORTALTEMPEL_22");
			SC_GotPortalTempelWalkthroughKey = TRUE;
		};
	};
```

**Reply by dosinabox** [reply]
свитки аддона - магия воды, не?

**Reply by Kor-Angar** [reply]
там нет разницы что они воды. маги воды учат же всем заклинаниям и огня и воды. и в продаже у них свитки все и воды и огня и призыва и нейтральные.

Ровно как и по воде у магов огня есть отдельный маг. то есть в монастыре есть маг огня специалист по воде, есть маг огня специалист по огню, есть маг огня специалист по призыву(по сути некромант). свитки магов воды есть в продаже в монастыре , по моему буря и в городе, то есть это не какой то эксклюзив или тайные знания.

пишу же я это касательно конкретной ситуации когда при игре за мага я подзапутался и бегал между мирами , так как не хватало ингредиентов и я уже решил в монастыре всё сделать, а там не оказалось что нет свитка. то есть это ситуация игровая. опять же таки исходя из того что я не бегал с одним спелом , а начал делать несколько рун. вот и сложилась подобная ситуация которую я тут и пописал.

---

я бы вообще не заморачивался а вручал один свиток при изучение схемы заклинания, что было бы весьма логично и понятно. // это отскакивая в сторону, мысли в слух.

**Reply by dosinabox** [reply]
> я бы вообще не заморачивался а вручал один свиток при изучение схемы заклинания

это очень муторно и сложно, если обучение происходит в другом мире

**Reply by Kor-Angar** [reply]
есть очень простой вариант который фактически не повлияет на игру за мага:
там где идёт проверка на входе в хоринос
```D
		if(NW_KDW_Removed_Forever == FALSE)
		{
```
добавить свитки в продажу Гораксу. просто по одному свитку магов воды из аддона для мага огня без всяких доп условий и проверок . вреда от такого не будет. эти свитки на баланс повлиять не смогут. 
я думаю что этим пираньи не сильно озаботились. особенно учитывая что баланс за магов вообще особо никто не дорабатывал. я собственно по этому и остановился в игре за них, так как надо время что бы выдохнуть.

**Reply by Kor-Angar** [reply]
и тут же захотелось поиграть в классику. что бы прочувствовать что маг огня это маг огня, а ни чмо из под нар :-)

**Reply by Kor-Angar** [reply]
> я бы вообще не заморачивался а вручал один свиток при изучение схемы заклинания, что было бы весьма логично и понятно. /
>> это очень муторно и сложно, если обучение происходит в другом мире

имел ввиду что  учитель учит героя и даёт ему один свиток этого заклинания :-)

**Comment by Kor-Angar** [comment]
у мага огня перед монастырём сильное кольцо(мана + жизнь) в продаже, а у мага огня внутри монастыря слабое кольцо маны. можно их поменять местами.

**Comment by Kor-Angar** [comment]
у меня Агон почему то работал в саду когда ГГ стал магом огня
![20230819210656_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/b664d17d-2af6-42f9-88df-09e96c18ad81)

**Reply by Kor-Angar** [reply]
```D
if(!Npc_IsDead(Agon))
	{
		if(Npc_KnowsInfo(other,DIA_Agon_GolemDead) || Npc_KnowsInfo(other,DIA_Agon_GolemLives))
		{
			B_RemoveNpc(NOV_603_Agon);
			Wld_InsertNpc(VLK_4007_Agon,"CITY2");
		}
		else
		{
			B_StartOtherRoutine(Nov607,"Start");
			B_StartOtherRoutine(Agon,"Start");
			Agon.aivar[AIV_DropDeadAndKill] = FALSE;
			Agon.aivar[AIV_NewsOverride] = FALSE;
			Agon.aivar[AIV_IgnoresArmor] = FALSE;
			Agon.aivar[AIV_IgnoresFakeGuild] = FALSE;
		};
	};
	AI_Output(self,other,"DIA_Pyrokar_OATH_11_08");	//Теперь, когда ты был принят в наши ряды, ты можешь поговорить с лордом Хагеном, главнокомандующим паладинов.
	AI_Output(self,other,"DIA_Pyrokar_OATH_11_09");	//Нам также очень интересно знать, как он оценивает ситуацию. Так что ты теперь можешь отправляться в Хоринис.
	AI_Output(self,other,"DIA_Pyrokar_OATH_11_10");	//Мы ожидаем, что ты принесешь его ответ немедленно.
	Player_KnowsLordHagen = TRUE;
```

**Reply by Kor-Angar** [reply]
в целом не понятно при каких условиях по твоей задумке Агон должен вернуться арбайтен в сад ;-) я имею ввиду с точки зрения ЛОРа игры.

**Reply by dosinabox** [reply]
В DIA_Agon_GolemDead и DIA_Agon_GolemLives Агон атакует ГГ с целью убить. После такого в монастырь он не вернется.

**Reply by Kor-Angar** [reply]
вопрос звучал несколько иначе ;-)

**Reply by Kor-Angar** [reply]
в целом не важно. твои проверки несработали.

**Reply by dosinabox** [reply]
странно, не вижу ошибок

**Reply by Kor-Angar** [reply]
я тоже не увидел. но в целом возвращение племянника в монастырь для того что бы заниматься садом это ошибка. к тому же в таком случае тебе нужно прорабатывать сценарий при котором Бабо становится садовником.

**Reply by Kor-Angar** [reply]
напомню что в MDK он должен был тусить в таверне, а в моей сборке он шёл к дяди в ратушу(это слово склоняется?).

**Reply by Kor-Angar** [reply]
> в целом не понятно при каких условиях по твоей задумке Агон должен вернуться арбайтен в сад ;-) я имею ввиду с точки зрения > ЛОРа игры.

Герой забрал руну не разговаривая с Агоном
https://www.youtube.com/live/fPX97chaM6Q?si=tYuu2tpVK_SQlNu-&t=2422

**Comment by Kor-Angar** [comment]
после вступления в маги огня, герою не обнулили преступления
![20230819210208_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/050441e6-21b6-4d9b-8ffa-8b3921245401)

**Reply by dosinabox** [reply]
Их обнуляет Парлан, только послушнику и только один раз.

**Reply by Kor-Angar** [reply]
ну и ладно.

**Comment by Kor-Angar** [comment]
в случае возвращения Игараца в монастырь они стоят уже в троём
![20230824064304_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/b39fa8ea-a76f-4d46-a1f1-c37c77b0aa83)
![20230824074829_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/946e4f4b-cad3-4160-a686-a7c5ed0ef241)

**Comment by Kor-Angar** [comment]
вот этот сундук слева разве не был открыт в оригинале? сейчас его надо взламывать 

![20230824064339_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/2bc68a02-ad69-43d7-a9c1-ed389040a46f)

**Reply by dosinabox** [reply]
Сундук в оригинале не заперт, но ему назначена комбинация и на модели висит замок. Было решено, что он не заперт по ошибке.
![Screenshot_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/39790055/803e7420-bedf-4012-936b-993f299ba8eb)
То же самое с сундуком в соседней комнате:
![Screenshot_2](https://github.com/dosinabox/g2nr_unofficial_update/assets/39790055/8850331e-d89b-4830-99d3-436c0c81b843)

**Comment by Kor-Angar** [comment]
@dosinabox  а что делает Агон в доме у судьи? :-)

**Comment by Kor-Angar** [comment]
29 минута. Ульф следует за Героем, при этом агрессивно реагирует на героя если тот достал оружие https://www.youtube.com/watch?v=fPX97chaM6Q&list=PLqe88GB0Po9yVjZy7u-oWBMVZufPC2kth&index=3

**Reply by Kor-Angar** [reply]
https://www.youtube.com/live/fPX97chaM6Q?si=yh8ZlLiNOnLQOobn&t=1753

**Reply by Kor-Angar** [reply]
![20230824055021_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/97b0d30f-637d-4452-bda9-5497ce1a99d9)

**Comment by Kor-Angar** [comment]
Бандит с агрился на героя, после того как героя укусила крыса https://www.youtube.com/live/fPX97chaM6Q?si=8QJ00LNvR0qhRsZW&t=1663
